google/compute_operation.go

```diff
@@ -74,93 +80,83 @@ type ComputeOperationError compute.OperationError
 		Project: project,
-		Type:    ComputeOperationWaitGlobal,
 	}
 
-	state := w.Conf()
-	state.Delay = 10 * time.Second
-	state.Timeout = time.Duration(timeoutMin) * time.Minute
-	state.MinTimeout = 2 * time.Second
-	opRaw, err := state.WaitForState()
-	if err != nil {
-		return fmt.Errorf("Error waiting for %s: %s", activity, err)
-	}
-
-	op = opRaw.(*compute.Operation)
-	if op.Error != nil {
-		return ComputeOperationError(*op.Error)
-	}
+	return waitComputeOperationWaiter(w, timeoutMin, activity)
+}
 
-	return nil
+func computeOperationWaitGlobal(config *Config, op *compute.Operation, project, activity string) error {
+	return computeOperationWaitGlobalTime(config, op, project, activity, 4)
 }
 
-func computeOperationWaitRegion(config *Config, op *compute.Operation, project string, region, activity string) error {
+func computeOperationWaitGlobalTime(config *Config, op *compute.Operation, project, activity string, timeoutMin int) error {
 	w := &ComputeOperationWaiter{
 		Service: config.clientCompute,
 		Op:      op,
 		Project: project,
-		Type:    ComputeOperationWaitRegion,
-		Region:  region,
 	}
 
-	state := w.Conf()
-	state.Delay = 10 * time.Second
-	state.Timeout = 4 * time.Minute
-	state.MinTimeout = 2 * time.Second
-	opRaw, err := state.WaitForState()
-	if err != nil {
-		return fmt.Errorf("Error waiting for %s: %s", activity, err)
-	}
+	return waitComputeOperationWaiter(w, timeoutMin, activity)
+}
 
-	op = opRaw.(*compute.Operation)
-	if op.Error != nil {
-		return ComputeOperationError(*op.Error)
+func computeOperationWaitRegion(config *Config, op *compute.Operation, project string, region, activity string) error {
+	return computeOperationWaitRegionTime(config, op, project, region, 4, activity)
+}
+
+func computeOperationWaitRegionTime(config *Config, op *compute.Operation, project, region string, timeoutMin int, activity string) error {
+	w := &ComputeOperationWaiter{
+		Service: config.clientCompute,
+		Op:      op,
+		Project: project,
 	}
 
-	return nil
+	return waitComputeOperationWaiter(w, timeoutMin, activity)
 }
 
-func computeOperationWaitZone(config *Config, op *compute.Operation, project string, zone, activity string) error {
+func computeOperationWaitZone(config *Config, op *compute.Operation, project, zone, activity string) error {
 	return computeOperationWaitZoneTime(config, op, project, zone, 4, activity)
 }
 
-func computeOperationWaitZoneTime(config *Config, op *compute.Operation, project string, zone string, minutes int, activity string) error {
+func computeOperationWaitZoneTime(config *Config, op *compute.Operation, project, zone string, timeoutMin int, activity string) error {
 	w := &ComputeOperationWaiter{
 		Service: config.clientCompute,
 		Op:      op,
 		Project: project,
-		Zone:    zone,
-		Type:    ComputeOperationWaitZone,
 	}
+
+	return waitComputeOperationWaiter(w, timeoutMin, activity)
+}
+
+func waitComputeOperationWaiter(w *ComputeOperationWaiter, timeoutMin int, activity string) error {
```

Comment:
I don't have a great suggestion for you but I don't like that "wait" occurs twice in this fn name. `waitCompute`? `waitComputeOperation` could work but I don't like that there's a `computeOperationWait` already. Or we can just leave it be since I don't think it'll need to be a separate fn once all the others are gone.